activerecord/lib/active_record/relation/query_methods.rb

```diff
@@ -1195,6 +1251,37 @@ def build_select(arel)
         end
       end
 
+      def build_with(arel)
+        return if with_values.empty?
+
+        is_recursive = with_values.delete(:recursive)
```

Comment:
I would prefer just `recursive` for this variable if you'll decide to keep recursive in this form of API.

Reply:
:+1: Changed to `recursive`